alluxio/worker_ring.py

```diff
@@ -170,30 +173,32 @@ def _get_etcd_client(self):
 class ConsistentHashProvider:
     def __init__(
         self,
-        worker_addresses,
+        etcd_hosts=None,
+        worker_hosts=None,
         logger=None,
+        options=None,
         num_virtual_nodes=2000,
         max_attempts=100,
+        refresh_timeout_seconds=120,
     ):
-        if not worker_addresses:
-            raise ValueError(
-                "'worker_addresses' must be provided to form worker hash ring."
-            )
-        self.num_virtual_nodes = num_virtual_nodes
-        self.max_attempts = max_attempts
-        self.logger = logger or logging.getLogger("ConsistentHashProvider")
-        self.is_ring_initialized = False
-        # init worker hash ring
-        hash_ring = SortedDict()
-        weight = math.ceil(self.num_virtual_nodes / len(worker_addresses))
-        for worker_address in worker_addresses:
-            worker_string = worker_address.dump_main_info()
-            for i in range(weight):
-                hash_key = self._hash(worker_string, i)
-                hash_ring[hash_key] = worker_address
-        self.worker_addresses = worker_addresses
-        self.hash_ring = hash_ring
-        self.is_ring_initialized = True
+        self._etcd_hosts = etcd_hosts
+        self._worker_addresses = (
+            WorkerNetAddress.from_worker_hosts(worker_hosts)
+            if worker_hosts
+            else None
+        )
+        self._num_virtual_nodes = num_virtual_nodes
+        self._max_attempts = max_attempts
+        self._logger = logger or logging.getLogger("ConsistentHashProvider")
+
+        self._lock = threading.Lock()
+        self._is_ring_initialized = False
+        self._update_hash_ring_internal(self._worker_addresses)
```

Comment:
@ssyssy please follow https://github.com/LuQQiu/alluxio-py/commit/33326c25ca227efc6521d8a8b139625ad087d8ef to modify here
because here won't do anything because 

def _update_hash_ring_internal(
        self, worker_addresses: Set[WorkerNetAddress]
    ):
        with self._lock:
            if worker_addresses == self._worker_addresses:
                return